packages/nimble-blazor/NimbleBlazor/Components/NimbleWaferMap.razor.cs

```diff
@@ -0,0 +1,179 @@
+    /// Represents the X coordinate of the maximum corner of the the grid bounding box for rendering the wafer map.
+    /// </summary>
+    [Parameter]
+    public double? GridMinY { get; set; }
+
+    /// <summary>
+    /// Represents the Y coordinate of the maximum corner of the the grid bounding box for rendering the wafer map.
+    /// </summary>
+    [Parameter]
+    public double? GridMaxY { get; set; }
+
+    /// <summary>
+    /// Represents the orientation of the notch on the wafer map outline.
+    /// </summary>
+    [Parameter]
+    public WaferMapOrientation? Orientation { get; set; }
+
+    /// <summary>
+    /// Represents the number of characters allowed to be displayed within a single die, including the label suffix.
+    /// </summary>
+    [Parameter]
+    public double? MaxCharacters { get; set; }
+
+    /// <summary>
+    /// Represents a boolean value that determines if the die labels in the wafer map view are shown or not. Default value is false.
+    /// </summary>
+    [Parameter]
+    public bool? DieLabelsHidden { get; set; }
+
+    /// <summary>
+    /// Represents a string that can be added as a label in the end of the each data information in the wafer map dies value.
+    /// </summary>
+    [Parameter]
+    public string? DieLabelsSuffix { get; set; }
+
+    /// <summary>
+    /// Represents an Enum value that determent if the colorScale is represent a continues gradient values (linear), or is set categorically (ordinal).
+    /// </summary>
+    [Parameter]
+    public WaferMapColorScaleMode? ColorScaleMode { get; set; }
+
+    /// <summary>
+    /// Represents a list of strings of dies that will be highlighted in the wafer map view.
+    /// </summary>
+    [Parameter]
+    public IEnumerable<string>? HighlightedTags
+    {
+        get
+        {
+            return _highlightedTags;
+        }
+        set
+        {
+            _highlightedTags = value;
+            _highlightedTagsUpdated = true;
+        }
+    }
+
+    /// <summary>
+    /// Represents the input data, an array of `WaferMapDie`, which fills the wafer map with content
+    /// </summary>
+    [Parameter]
+    public IEnumerable<WaferMapDie>? Dies
+    {
+        get
+        {
+            return _dies;
+        }
+        set
+        {
+            _dies = value;
+            _diesUpdated = true;
+        }
+    }
+
+    /// <summary>
+    /// Represents the color spectrum which shows the status of the dies on the wafer.
+    /// </summary>
+    [Parameter]
+    public WaferMapColorScale? ColorScale
+    {
+        get
+        {
+            return _colorScale;
+        }
+        set
+        {
+            _colorScale = value;
+            _colorScaleUpdated = true;
+        }
+    }
+
+    /// <summary>
+    /// Will be triggered to inform the user that the state of hovering over a die has changed.
+    /// </summary>
+    [Parameter]
+    public EventCallback<WaferMapHoverDieChangedEventArgs> HoverDieChanged { get; set; }
```

Comment:
It doesn't look like this event is hooked up correctly right now. In order to use Blazor's custom event bindings, in addition to what you have, you also need:
- NimbleBlazor.lib.module.js needs a call to `Blazor.registerCustomEventType()` for your event. The 1st arg is `nimblewafermapdiehoverchange` (note: without the `on` at the beginning), the browserEventName is what the Nimble component event name is (looks like `die-hover` for the wafer map), and you need to pass along the relevant parts of the event args (see the other calls in that file for an example)
- `NimbleWaferMap.razor` needs something like
```html
    @onnimblewafermapdiehoverchange="(__value) => HandleDieHoverChange(__value)"
```
then `HandleDieHoverChange` needs to trigger your `HoverDieChanged` `EventCallback`.

Since there's NimbleBlazor-specific JavaScript executing for the custom event code path, optimally you could add an acceptance test to exercise that too. Playwright has `hover()` on elements as well as `page.mouse` APIs that you could try using to hover over the wafer map. Then the acceptance test Razor file could listen to `HoverDieChanged` and toggle a checkbox, or something like that.